From d2310e5c9793b832577a633e41743260b29d875b Mon Sep 17 00:00:00 2001
From: David Henningsson <david.henningsson@canonical.com>
Date: Wed, 14 Sep 2011 14:55:07 +0200
Subject: [PATCH] Make sure module-dbus-protocol is loaded after
 module-default-device-restore

Module-dbus-protocol gets the default sink, which sets the default sink
if not already set. This is turn makes module-default-device-restore do
nothing.
To solve the problem, make sure module-default-device-restore is loaded
before module-dbus-protocol and not the other way around.

BugLink: http://bugs.launchpad.net/bugs/843780
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---
 src/daemon/default.pa.in |   12 +++++++-----
 1 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/daemon/default.pa.in b/src/daemon/default.pa.in
index 6155b31..dfecbc4 100755
--- a/src/daemon/default.pa.in
+++ b/src/daemon/default.pa.in
@@ -98,11 +98,6 @@ load-module module-esound-protocol-unix
 .endif
 load-module module-native-protocol-unix
 ])dnl
-ifelse(@HAVE_DBUS@, 1, [dnl
-.ifexists module-dbus-protocol@PA_SOEXT@
-load-module module-dbus-protocol
-.endif
-])dnl
 
 ### Network access (may be configured with paprefs, so leave this commented
 ### here if you plan to use paprefs)
@@ -165,6 +160,13 @@ load-module module-filter-heuristics
 load-module module-filter-apply
 ])dnl
 
+ifelse(@HAVE_DBUS@, 1, [dnl
+### Load DBus protocol
+.ifexists module-dbus-protocol@PA_SOEXT@
+load-module module-dbus-protocol
+.endif
+])dnl
+
 ifelse(@HAVE_X11@, 1, [dnl
 # X11 modules should not be started from default.pa so that one daemon
 # can be shared by multiple sessions.
-- 
1.7.5.4

